File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/model/storage/CardFilterStorage.java
package com.dbottillo.mtgsearchfree.model.storage;

import android.content.SharedPreferences;

import com.dbottillo.mtgsearchfree.model.CardFilter;
import com.dbottillo.mtgsearchfree.model.CardProperties;
import com.dbottillo.mtgsearchfree.util.LOG;

public class CardFilterStorage {

    SharedPreferences preferences;

    public CardFilterStorage(SharedPreferences preferences) {
        LOG.d("created");
        this.preferences = preferences;
    }

    public CardFilter load() {
        LOG.d();
        CardFilter res = new CardFilter();
        res.white = preferences.getBoolean(CardProperties.COLOR.WHITE.key, true);
        res.blue = preferences.getBoolean(CardProperties.COLOR.BLUE.key, true);
        res.black = preferences.getBoolean(CardProperties.COLOR.BLACK.key, true);
        res.red = preferences.getBoolean(CardProperties.COLOR.RED.key, true);
        res.green = preferences.getBoolean(CardProperties.COLOR.GREEN.key, true);

        res.artifact = preferences.getBoolean(CardProperties.TYPE.ARTIFACT.key, true);
        res.land = preferences.getBoolean(CardProperties.TYPE.LAND.key, true);
        res.eldrazi = preferences.getBoolean(CardProperties.TYPE.ELDRAZI.key, true);

        res.common = preferences.getBoolean(CardProperties.RARITY.COMMON.key, true);
        res.uncommon = preferences.getBoolean(CardProperties.RARITY.UNCOMMON.key, true);
        res.rare = preferences.getBoolean(CardProperties.RARITY.RARE.key, true);
        res.mythic = preferences.getBoolean(CardProperties.RARITY.MYTHIC.key, true);

        return res;
    }

    public void sync(CardFilter filter) {
        LOG.d();
        preferences.edit()
                .putBoolean(CardProperties.COLOR.WHITE.key, filter.white)
                .putBoolean(CardProperties.COLOR.BLUE.key, filter.blue)
                .putBoolean(CardProperties.COLOR.BLACK.key, filter.black)
                .putBoolean(CardProperties.COLOR.RED.key, filter.red)
                .putBoolean(CardProperties.COLOR.GREEN.key, filter.green)
                .putBoolean(CardProperties.TYPE.ARTIFACT.key, filter.artifact)
                .putBoolean(CardProperties.TYPE.LAND.key, filter.land)
                .putBoolean(CardProperties.TYPE.ELDRAZI.key, filter.eldrazi)
                .putBoolean(CardProperties.RARITY.COMMON.key, filter.common)
                .putBoolean(CardProperties.RARITY.UNCOMMON.key, filter.uncommon)
                .putBoolean(CardProperties.RARITY.RARE.key, filter.rare)
                .putBoolean(CardProperties.RARITY.MYTHIC.key, filter.mythic)
                .apply();
    }


}

File: mtgSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/model/storage/CardFilterStorage.kt
package com.dbottillo.mtgsearchfree.model.storage

import android.content.SharedPreferences
import com.dbottillo.mtgsearchfree.model.CardFilter
import com.dbottillo.mtgsearchfree.model.CardProperties

class CardFilterStorage(var preferences: SharedPreferences) {

    fun load(): CardFilter {
        var res = CardFilter()
        res.white = preferences.getBoolean(CardProperties.COLOR_WHITE, true)
        res.blue = preferences.getBoolean(CardProperties.COLOR_BLUE, true)
        res.black = preferences.getBoolean(CardProperties.COLOR_BLACK, true)
        res.red = preferences.getBoolean(CardProperties.COLOR_RED, true)
        res.green = preferences.getBoolean(CardProperties.COLOR_GREEN, true)

        res.artifact = preferences.getBoolean(CardProperties.TYPE_ARTIFACT, true)
        res.land = preferences.getBoolean(CardProperties.TYPE_LAND, true)
        res.eldrazi = preferences.getBoolean(CardProperties.TYPE_ELDRAZI, true)

        res.common = preferences.getBoolean(CardProperties.RARITY_COMMON, true)
        res.uncommon = preferences.getBoolean(CardProperties.RARITY_UNCOMMON, true)
        res.rare = preferences.getBoolean(CardProperties.RARITY_RARE, true)
        res.mythic = preferences.getBoolean(CardProperties.RARITY_MYHTIC, true)

        return res;
    }

    fun sync(filter: CardFilter) {
        preferences.edit()
                .putBoolean(CardProperties.COLOR_WHITE, filter.white)
                .putBoolean(CardProperties.COLOR_BLUE, filter.blue)
                .putBoolean(CardProperties.COLOR_BLACK, filter.black)
                .putBoolean(CardProperties.COLOR_RED, filter.red)
                .putBoolean(CardProperties.COLOR_GREEN, filter.green)
                .putBoolean(CardProperties.TYPE_ARTIFACT, filter.artifact)
                .putBoolean(CardProperties.TYPE_LAND, filter.land)
                .putBoolean(CardProperties.TYPE_ELDRAZI, filter.eldrazi)
                .putBoolean(CardProperties.RARITY_COMMON, filter.common)
                .putBoolean(CardProperties.RARITY_UNCOMMON, filter.uncommon)
                .putBoolean(CardProperties.RARITY_RARE, filter.rare)
                .putBoolean(CardProperties.RARITY_MYHTIC, filter.mythic)
                .apply()
    }


}

